backend/domain_types/src/types.rs

```
@@ -233,8 +233,188 @@ impl<
+                                    sub_code: "MISSING_APPLE_PAY_PAYMENT_METHOD".to_owned(),
+                                    error_identifier: 400,
+                                    error_message: "Apple Pay payment method is required".to_owned(),
+                                    error_object: None,
+                                })
+                            })?;
+
+                            let wallet_data = payment_method_data::ApplePayWalletData {
+                                payment_data: applepay_payment_data,
+                                payment_method: payment_method_data::ApplepayPaymentMethod {
+                                    display_name: payment_method.display_name,
+                                    network: payment_method.network,
+                                    pm_type: payment_method.r#type,
+                                },
+                                transaction_identifier: apple_wallet.transaction_identifier,
+                            };
+                            Ok(PaymentMethodData::Wallet(payment_method_data::WalletData::ApplePay(wallet_data)))
+                        }
+                        Some(grpc_api_types::payments::wallet_payment_method_type::WalletType::GooglePay(google_wallet)) => {
+                            let info = google_wallet.info.ok_or_else(|| {
+                                ApplicationErrorResponse::BadRequest(ApiError {
+                                    sub_code: "MISSING_GOOGLE_PAY_INFO".to_owned(),
+                                    error_identifier: 400,
+                                    error_message: "Google Pay payment method info is required".to_owned(),
+                                    error_object: None,
+                                })
+                            })?;
+
+                            let tokenization_data = google_wallet.tokenization_data.ok_or_else(|| {
```

Use change context to lift the error don't discard the errors

---

Root error